### PR TITLE
Allow blacklisting of suites and tests for specific entities

### PR DESCRIPTION
**Pull request by rjkip**

### CODE REVIEW

**Comment by rjkip** [comment]
@DRvanR

File: src/Surfnet/Conext/EntityVerificationFramework/Tests/SuiteTest.php [hunk 1]
```diff
@@ -144,5 +176,27 @@ private function getMockContext()
 
         return $context;
     }
+
+    /**
+     * @return MockInterface|VerificationBlacklist
+     */
+    private function createBlacklistDummy()
```
Comment (DRvanR) [inline src/Surfnet/Conext/EntityVerificationFramework/Tests/SuiteTest.php]: `getMockBlacklist`? As that is what you're getting :wink:

File: src/Surfnet/Conext/OperationsSupportBundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -47,6 +50,37 @@ public function getConfigTreeBuilder()
                         ->end()
                     ->end()
                 ->end()
+                ->arrayNode('blacklist')
+                    ->info('Blacklist tests or entire suites for specific entities (suite, suite.test, *)')
+                    ->performNoDeepMerging()
+                    ->validate()
+                        ->always(function ($suitesOrTests) {
+                            foreach (array_keys($suitesOrTests) as $suiteOrTestName) {
+                                if ($suiteOrTestName === '*') {
```
Comment (DRvanR) [inline src/Surfnet/Conext/OperationsSupportBundle/DependencyInjection/Configuration.php]: use constant

File: src/Surfnet/Conext/EntityVerificationFramework/VerificationBlacklistFactory.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+     * @param array[] $entityDescriptors
+     * @return EntitySet
+     */
+    private static function createEntitySetFromEntityDescriptors(array $entityDescriptors)
+    {
+        Assert::allIsArray($entityDescriptors, 'Entity descriptors must consist of arrays');
+
+        return new EntitySet(
+            array_map(
+                function (array $entityDescriptor) {
```
Comment (DRvanR) [inline src/Surfnet/Conext/EntityVerificationFramework/VerificationBlacklistFactory.php]: as discussed: Seeing this function also does validation, not just map value to value, it deserves to be more than a callback :wink:

**Comment by DRvanR** [comment]
As discussed, would consider merging the wildcard EntitySet with the other EntitySets.

File: src/Surfnet/Conext/EntityVerificationFramework/Tests/SuiteTest.php (hunk 1 above)
Comment (rjkip) [inline src/Surfnet/Conext/EntityVerificationFramework/Tests/SuiteTest.php]: I'm creating a blacklist ~~dummy~~ stub :(

**Comment by rjkip** [comment]
@DRvanR

**Comment by DRvanR** [comment]
Little nit: might want to document what an EntityDescriptor is, other than that :+1:

**Pull request merged**